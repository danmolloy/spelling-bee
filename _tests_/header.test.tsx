import { render, screen, act, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom"
import Header from "../components/header";

const mockProps = {
  date: "mock Date",
  editor: "mock Editor",
  setShowMenu: jest.fn(),
  showMenu: false
}

describe("Header component", () => {
  beforeEach(() => {
    render(<Header {...mockProps} />)
  })
  it("Renders", () => {
    const header = screen.getByTestId("header-div")
    expect(header).toBeInTheDocument()
  })
  it("Game title is in the document", () => {
    const header = screen.getByTestId("header-div")
    expect(header.textContent).toMatch("Spelling Bee")
  })
  it("Game date is in the document", () => {
    const header = screen.getByTestId("header-div")
    expect(header.textContent).toMatch(`${mockProps.date} (Yesterday)`)
  })
  it("Game editor's name is in the document", () => {
    const header = screen.getByTestId("header-div")
    expect(header.textContent).toMatch(`Original NYT game edited by ${mockProps.editor}`)
  })
  it("Daniel Molloy's name is in the document", () => {
    const header = screen.getByTestId("header-div")
    expect(header.textContent).toMatch(`Replicated by Daniel Molloy`)
  })
  it("Menu button is in the document", () => {
    const menuIcon = screen.getByTestId("menu-icon")
    expect(menuIcon).toBeInTheDocument()
  })
  it("Menu button calls setShowMenu on click", () => {
    const menuIcon = screen.getByTestId("menu-icon")
    act(() => {
      fireEvent.click(menuIcon)
    })
    expect(mockProps.setShowMenu).toBeCalled()
  })
})