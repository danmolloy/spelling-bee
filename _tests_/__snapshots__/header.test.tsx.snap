// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<div
  class="md:p-4 flex flex-row md:justify-between w-full md:items-start justify-end "
  data-testid="header-div"
>
  <div
    class="hidden md:flex flex-col"
  >
    <div
      class="flex flex-row items-end"
    >
      <h1
        class="font-display font-bold text-4xl pr-4"
      >
        Spelling Bee
      </h1>
      <h2
        class="font-thin"
      >
        mockDate
         (Yesterday)
      </h2>
    </div>
    <div>
      <h2>
        Original NYT game edited by 
        mockEditor
      </h2>
      <h2>
        Replicated by Daniel Molloy
      </h2>
    </div>
  </div>
  <div
    class="flex flex-col items-end"
  >
    <button
      class="hover:bg-gray-100 active:bg-gray-200 text-2xl m-2 w-10 h-10 rounded-full flex items-center justify-center"
      data-testid="menu-icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
        />
      </svg>
    </button>
  </div>
</div>
`;
